pkgs/development/compilers/rust/rustc.nix

```diff
@@ -57,27 +52,67 @@ stdenv.mkDerivation {
 
   # We need rust to build rust. If we don't provide it, configure will try to download it.
   # Reference: https://github.com/rust-lang/rust/blob/master/src/bootstrap/configure.py
-  configureFlags = configureFlags
-                ++ [ "--enable-local-rust" "--local-rust-root=${rustPlatform.rust.rustc}" "--enable-rpath"
-                     "--enable-vendor"
-                     "--default-linker=${targetPackages.stdenv.cc}/bin/cc" ]
-                ++ optional (!withBundledLLVM) [ "--enable-llvm-link-shared" "--llvm-root=${llvmShared}" ]
-                ++ optional (targets != []) "--target=${target}";
+  configureFlags = let
+    setHost   = "--set=target.${stdenv.hostPlatform.config}";
+    setBuild  = "--set=target.${stdenv.buildPlatform.config}";
+    setTarget = "--set=target.${stdenv.targetPlatform.config}";
+    ccForHost  = "${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc";
+    cxxForHost = "${stdenv.cc}/bin/${stdenv.cc.targetPrefix}c++";
+    ccForBuild  = "${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}cc";
+    cxxForBuild = "${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}c++";
+    ccForTarget  = "${targetPackages.stdenv.cc}/bin/${targetPackages.stdenv.cc.targetPrefix}cc";
+    cxxForTarget = "${targetPackages.stdenv.cc}/bin/${targetPackages.stdenv.cc.targetPrefix}c++";
+  in [
+    "--release-channel=stable"
+    "--set=build.rustc=${rustPlatform.rust.rustc}/bin/rustc"
+    "--set=build.cargo=${rustPlatform.rust.cargo}/bin/cargo"
+    "--enable-rpath"
+    "--enable-vendor"
+    "--build=${stdenv.buildPlatform.config}"
+    "--host=${stdenv.hostPlatform.config}"
+    "--target=${stdenv.targetPlatform.config}"
+
+    "${setBuild}.cc=${ccForBuild}"
+    "${setHost}.cc=${ccForHost}"
+
+    "${setBuild}.linker=${ccForBuild}"
+    "${setHost}.linker=${ccForHost}"
+
+    "${setBuild}.cxx=${cxxForBuild}"
+    "${setHost}.cxx=${cxxForHost}"
+  ] ++ optional (!withBundledLLVM) [
+    "--enable-llvm-link-shared"
+    "${setBuild}.llvm-config=${buildLlvmShared}/bin/llvm-config"
+    "${setHost}.llvm-config=${buildLlvmShared}/bin/llvm-config"
```

Comment:
this might be wrong, need to test

Reply:
so when building under ARMv7 rust tries to call `llvm-config` to get the libs to link against. `buildLlvmShared` has an `llvm-config` that can run on the build system but the wrong shared libraries, while `llvm-config` is the opposite (wrong binary, right libs). Not sure why this problem doesn't show up on AArch64.

Reply:
Maybe a shim `llvm-config` script should be introduced, this is similar to meta-amd's: https://git.yoctoproject.org/cgit/cgit.cgi/meta-amd/tree/common/recipes-core/llvm/files/0001-llvm-config-allow-overriding-libdir-through-cmdline.patch?h=sumo

Reply:
Since there should be no arch-specific one could wrap around the llvm-config
and then just do `sed -e "s!${buildPackages.llvm}!${llvm}!" on the output?